src/base/memory_monitor_sr.cpp

```diff
@@ -402,29 +485,116 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
+  }
+
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO *&info, int module_index) const
+  {
+    int idx = 0;
+    int error = NO_ERROR;
+    if (module_index == -1)
+      {
+	// aggregate all detail memory information of modules
+	for (idx; idx < MMON_MODULE_LAST; idx++)
+	  {
+	    error = m_mmon->m_module[idx]->aggregate_stats (info[idx], false);
+	    if (error != NO_ERROR)
+	      {
+		return error;
+	      }
+	  }
+      }
+    else
+      {
+	error =  m_mmon->m_module[module_index]->aggregate_stats (info[idx], false);
+      }
+
+    return error;
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  int memory_monitor::aggregater::get_module_info_summary (MMON_MODULE_INFO *&info, bool sorted_result) const
   {
-    return 0;
+    int error = NO_ERROR;
+
+    for (int i = 0; i < MMON_MODULE_LAST; i++)
+      {
+	error = m_mmon->m_module[i]->aggregate_stats (info[i], true);
+	if (error != NO_ERROR)
+	  {
+	    return error;
+	  }
+      }
+
+    if (sorted_result)
+      {
+	const auto &comp = [] (const auto& module_info1, const auto& module_info2)
+	{
+	  return module_info1.stat.cur_stat > module_info2.stat.cur_stat;
+	};
+	std::sort (info, info + MMON_MODULE_LAST, comp);
+      }
+
+    return error;
   }
 
-  int memory_monitor::aggregater::get_transaction_info (const MMON_TRAN_INFO &info, int tran_count)
+  int memory_monitor::aggregater::get_transaction_info (MMON_TRAN_INFO &info, int tran_count) const
   {
-    return 0;
+    int num_total_indices = 0;
+    LOG_TDES **trantable;
+    LOG_TDES *tdes = nullptr;
+    std::vector <std::pair <int, uint64_t>> tran_info;
+
+    logtb_get_tran_memory_info_nolatch (tran_info);
+
+    if (tran_count == 0)
+      {
+	// transaction option with no tran-count option
+	info.num_tran = tran_info.size ();
+      }
+    else
+      {
+	// 1) transaction option with tran-count option
+	// 2) default option (cubrid memmon <server-name>)
```

Comment:
We shouldn't mention the "default" here.
We can describe here that It's just get the max value between the request tran count and the active tran count. I think we even don't need the comment with the `std::max`.
 ``` info.num_tran = std::max (tran_count, (int) tran_info.size ());```

Reply:
I agree. But I think we have to use `std::min`.

Reply:
:)